homeassistant/components/deconz/binary_sensor.py

```diff
@@ -193,4 +211,5 @@ def async_update_callback(self) -> None:
     @property
     def is_on(self) -> bool:
         """Return the state of the sensor."""
+        assert isinstance(self._device.tampered, bool)
```

Comment:
Possible change in behaviour? [Raw state](https://github.com/Kane610/deconz/blob/e5f3a3a95a62a276bd9d7118683f416cfb3d263f/pydeconz/sensor.py#L178) can return None, so is assert appropriate here?

Reply:
Yes that is true, but the entity would only be created [if it's not None](https://github.com/home-assistant/core/blob/9769a8c08e68a0c616004f651bbc36a6a44505f1/homeassistant/components/deconz/binary_sensor.py#L97)

Reply:
Ok fair enough, maybe a comment should explain why this assert won't fail in normal usage.

Reply:
I've removed the asserts based on feedback in another PR